src/ui/movable_rect.hpp

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <types.hpp>
+
+namespace nebula {
+  /**
+   * Movable rectangle ( in short MR ) is movable visual copy of MR Button
+   *created
+   *after interaction with it. This structure holds information about objects
+   *current location in the application window and pointer to render function
+   *inherited from MR Button.
+   **/
+  using Movable_Rect = struct Movable_Rect {
+    /** window_width - window width. Important for correct position
+     *                       calculation during movement
+     */
+    static f32 window_height;
+    /** window_height - window height. Important for correct position
+     *                        calculation during movement
+     */
+    static f32 window_width;
+
+    void (*render_function)(anton::Rect_f32 position);
+    anton::Rect_f32 position;
```

Comment:
how is position a rect?

Reply:
Top left corner (x1, y1) , Top right (x2, y1), Bottom left (x1, y2), Bottom right ( x2, y2). Rect_f32 = {.left = x1, .right = x2, .top = y1, .bottom = y2}